src/chunkserver/copyset_node.cpp

```diff
@@ -1001,38 +1002,60 @@ void CopysetNode::ForceSyncAllChunks() {
     isSyncing_ = false;
 }
 
+void CopysetNode::setSharedSyncPool(
+    const std::shared_ptr<TaskThreadPool<>> &threadpool) {
+    copysetSyncPool_ = threadpool;
+}
+
 void CopysetNode::SyncAllChunks() {
     std::deque<ChunkID> temp;
     {
         curve::common::LockGuard lg(chunkIdsLock_);
         temp.swap(chunkIdsToSync_);
     }
-    size_t total = temp.size();
     std::set<ChunkID> chunkIds;
     for (auto chunkId : temp) {
         chunkIds.insert(chunkId);
     }
     for (ChunkID chunk : chunkIds) {
-        CSErrorCode r = dataStore_->SyncChunk(chunk);
-        if (r != CSErrorCode::Success) {
-            LOG(FATAL) << "Sync Chunk failed in Copyset: "
+        copysetSyncPool_->Enqueue([=]() {
+            CSErrorCode r = dataStore_->SyncChunk(chunk);
+            if (r != CSErrorCode::Success) {
+                LOG(FATAL) << "Sync Chunk failed in Copyset: "
                        << GroupIdString()
                        << ", chunkid: " << chunk
                        << " data store return: " << r;
-        }
+            }
+        });
     }
 }
 
-int SyncTimer::init(CopysetNode *node, int timeoutMs) {
-    if (RepeatedTimerTask::init(timeoutMs) != 0) {
-        return -1;
-    }
+void SyncChunkThread::Init(CopysetNode* node) {
+    running_ = true;
     node_ = node;
-    return 0;
+    cond_ = std::make_shared<std::condition_variable>();
+}
+
+void SyncChunkThread::Run() {
+    syncThread_ = std::thread([this](){
+        while (running_) {
+            std::unique_lock<std::mutex> lock(mtx_);
+            cond_->wait_for(lock, std::chrono::seconds(30));
+            node_->ForceSyncAllChunks();
```

Comment:
Use "SyncAllChunks()" other than "ForceSyncAllChunks()" may be better,  we need sync chunk file immediately here, but "ForceSyncAllChunks()" will sleep 500ms by default.

Reply:
> Use "SyncAllChunks()" other than "ForceSyncAllChunks()" may be better, we need sync chunk file immediately here, but "ForceSyncAllChunks()" will sleep 500ms by default.

It seems right.